sw/airborne/firmwares/rotorcraft/stabilization/stabilization_rate_indi.c

```diff
@@ -22,38 +23,194 @@
 /** @file stabilization_rate_indi.c
  *  Rate stabilization for rotorcrafts based on INDI by Ewoud Smeur.
  */
+
 #include "firmwares/rotorcraft/stabilization.h"
-#include "firmwares/rotorcraft/stabilization/stabilization_rate.h"
+#include "firmwares/rotorcraft/stabilization/stabilization_rate_indi.h"
 
 #ifdef STABILIZATION_ATTITUDE_INDI_SIMPLE
 #include "firmwares/rotorcraft/stabilization/stabilization_indi_simple.h"
 #else
 #include "firmwares/rotorcraft/stabilization/stabilization_indi.h"
 #endif
 
+/* -- define variables */
+struct FloatRates stabilization_rate_sp;
+struct ReferenceSystem reference_acceleration = {
+  STABILIZATION_INDI_REF_ERR_P,
+  STABILIZATION_INDI_REF_ERR_Q,
+  STABILIZATION_INDI_REF_ERR_R,
+  STABILIZATION_INDI_REF_RATE_P,
+  STABILIZATION_INDI_REF_RATE_Q,
+  STABILIZATION_INDI_REF_RATE_R
+}; // these values are defined in airframe.h file
+
+// variables in case we use "stabilization_indi.c" 
+float q_filt = 0.0;
+float r_filt = 0.0;
+
+/* -- define functions */
+static void stabilization_indi_rate_calc_cmd(int32_t indi_commands[], struct FloatRates rate_ref, bool in_flight)
+
+/* -- RC deadbands */ 
+#ifndef STABILIZATION_RATE_DEADBAND_P
+#define STABILIZATION_RATE_DEADBAND_P 0
+#endif
+#ifndef STABILIZATION_RATE_DEADBAND_Q
+#define STABILIZATION_RATE_DEADBAND_Q 0
+#endif
+#ifndef STABILIZATION_RATE_DEADBAND_R
+#define STABILIZATION_RATE_DEADBAND_R 200
+#endif
+
+#define ROLL_RATE_DEADBAND_EXCEEDED()                                         \
+  (radio_control.values[RADIO_ROLL] >  STABILIZATION_RATE_DEADBAND_P || \
+   radio_control.values[RADIO_ROLL] < -STABILIZATION_RATE_DEADBAND_P)
+
+#define PITCH_RATE_DEADBAND_EXCEEDED()                                         \
+  (radio_control.values[RADIO_PITCH] >  STABILIZATION_RATE_DEADBAND_Q || \
+   radio_control.values[RADIO_PITCH] < -STABILIZATION_RATE_DEADBAND_Q)
+
+#define YAW_RATE_DEADBAND_EXCEEDED()                                         \
+  (radio_control.values[RADIO_YAW] >  STABILIZATION_RATE_DEADBAND_R || \
+   radio_control.values[RADIO_YAW] < -STABILIZATION_RATE_DEADBAND_R)
+
+
+#if PERIODIC_TELEMETRY
+#include "subsystems/datalink/telemetry.h"
+
+static void send_rate(struct transport_tx *trans, struct link_device *dev)
+{
+  pprz_msg_send_RATE_LOOP(trans, dev, AC_ID,
+                          &stabilization_rate_sp.p,
+                          &stabilization_rate_sp.q,
+                          &stabilization_rate_sp.r);
```

Comment:
It seems that the `RATE_LOOP` message have more field than that: http://docs.paparazziuav.org/latest/paparazzi_messages.html#RATE_LOOP